src/foam/core/InvalidX.java

```diff
@@ -6,29 +6,43 @@
 
 package foam.core;
 
+import foam.nanos.logger.Logger;
+
 public class InvalidX
-  extends AbstractX
+  extends ProxyX
 {
-  private RuntimeException exception_;
+  private X delegate_;
+  private String nspecName_;
+  private Logger logger_;
 
-  public InvalidX(RuntimeException exception) {
-    exception_ = exception;
+  public InvalidX(X delegate, String nspecName, Logger logger) {
+    super(delegate);
+    delegate_ = delegate;
+    nspecName_ = nspecName;
+    logger_ = logger;
   }
 
+  @Override
   public Object get(X x, Object key) {
-    throw exception_;
+    Object ret = getX().get(x, key);
+    logger_.warning("Unsafe access to " + nspecName_ + ". Please use .inX() instead.");
+
+    return ret;
   }
 
+  @Override
   public X put(Object key, Object value) {
-    throw exception_;
-  }
+    setX(getX().put(key, value));
```

Comment:
Rather than copy this code from `ProxyX`, we should just change the line below from

```Java
  return (X) this;
```

to

```Java
  return super.put(key, value);
```